### PR TITLE
[stable/prometheus-operator] Sync prometheus operator dashboard and rules

### PR DESCRIPTION
**Pull request by ElfoLiNk**
Signed-off-by: Matteo Gazzetta <matteo_gazzetta@hotmail.it>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by zanhsieh** [comment]
@ElfoLiNk 
Could you bump the chart version to the latest one please? Thank you.

**Comment by vsliouniaev** [comment]
Could you please update from latest, the tests should work
/test all

**Comment by zanhsieh** [comment]
@ElfoLiNk @vsliouniaev 
I think e2e failed by unable to clean up CRD resources. What do you guys think?

**Review by vsliouniaev — APPROVED** [review]

**Comment by vsliouniaev** [comment]
/retest
Another CRD failure, Helm doesn't support CRDs 😡

**Comment by vsliouniaev** [comment]
Looks like this is happening https://github.com/helm/charts/issues/9241#issuecomment-545771389

I'm going to try solve this by using two separate CI test files, get the CRDs into the cluster idempotently using the operator pod, and leave the CRDs in there after the test completes

**Comment by vsliouniaev** [comment]
I doubt tg tests will pass before that PR gets in

**Comment by ElfoLiNk** [comment]
Had to remove PrometheusRemoteWriteDesiredShards rule due to {{ template issue

**Comment by vsliouniaev** [comment]
/ok-to-test

@ElfoLiNk These rules are not valid with the current version of the prometheus-operator image. I get
```
Error: UPGRADE FAILED: failed to create resource: admission webhook "prometheusrulemutate.monitoring.coreos.com" denied the request: Rules are not valid
```

- Updating the image to version 0.34.0 requires clearing out the unsupported -crd-apigroup flag.
- This also technically breaks the upgrade process for folks - they would have to upgrade to the 0.34.0 image first, then upgrade the rules, otherwise their release will fail.

We'll need to coordinate and do the following things:
1. I will do the image upgrade first, in a separate PR, so that a version of this chart exists that someone can upgrade to.
2. Doing a rules sync will require that we bump the major chart version, since the upgrade is no longer transparent and requires a minimum operator version.

File: stable/prometheus-operator/templates/prometheus/rules-1.14/kubernetes-system-apiserver.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Generated from 'kubernetes-system-apiserver' group from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/prometheus-rules.yaml
+# Do not change in-place! In order to change this file first read following link:
+# https://github.com/helm/charts/tree/master/stable/prometheus-operator/hack
+{{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
+{{- if and (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion) .Values.defaultRules.create }}
+apiVersion: {{ printf "%s/v1" (.Values.prometheusOperator.crdApiGroup | default "monitoring.coreos.com") }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/prometheus/rules-1.14/kubernetes-system-apiserver.yaml]: I see this `crdApiGroup` value has re-appeared. Could you please try re-sync the rules again?

**Review by vsliouniaev — CHANGES_REQUESTED** [review]
This still needs a section in the README.md file to describe the 7.x to the 8.x upgrade process:


### Upgrading from 7.x.x to 8.x.x
Due to new template functions being used in the rules in version `8.x.x` of the chart, an upgrade to Prometheus Operator and Prometheus is necessary in order to support them. First, upgrade to the latest version of `7.x.x`
```sh
helm upgrade <your-release-name> stable/prometheus-operator --version 7.4.0
```
Then upgrade to `8.x.x`
```sh
helm upgrade <your-release-name> stable/prometheus-operator 
```

**Pull request closed without merging**